chain-impl-mockchain/src/config.rs

```diff
@@ -99,75 +205,102 @@ impl ConfigParam for Discrimination {
     }
 }
 
-impl ConfigParam for ConsensusVersion {
-    const TAG: Tag = Tag::unchecked_new(3);
+impl ConfigParamVariant for ConsensusVersion {
+    const TAG: Tag = Tag::new(3);
     const NAME: &'static str = "block0-consensus";
 
-    fn to_payload(&self) -> TagPayload {
+    fn to_payload(&self) -> Vec<u8> {
         (*self as u16).to_be_bytes().to_vec()
     }
 
-    fn from_payload(payload: &TagPayload) -> Result<Self, Error> {
+    fn from_payload(payload: &[u8]) -> Result<Self, Error> {
         let mut bytes = 0u16.to_ne_bytes();
         if payload.len() != bytes.len() {
             return Err(Error::SizeInvalid);
         };
-        bytes.copy_from_slice(&payload);
+        bytes.copy_from_slice(payload);
         let integer = u16::from_be_bytes(bytes);
         ConsensusVersion::from_u16(integer).ok_or(Error::StructureInvalid)
     }
 
-    fn to_string(&self) -> String {
+    fn to_cfg_string(&self) -> String {
         format!("{}", self)
     }
 
-    fn from_string(s: &str) -> Result<Self, Error> {
+    fn from_cfg_str(s: &str) -> Result<Self, Error> {
         s.parse().map_err(|_| Error::UnknownString(s.to_string()))
     }
 }
 
-pub fn entity_to<T: ConfigParam>(t: &T) -> (Tag, TagPayload) {
-    (T::TAG, t.to_payload())
-}
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+struct Tag(u16);
```

Comment:
it would be better not to have moved code around (Tag/TagLen), it make reading the diff a bit harder.

Reply:
Tag and TagLen became private, so I wanted them not to overshadow public things, which are obviously more important for readers. I considered moving them to a private module in separate file to make project easier to navigate, but the team seems to prefer long, self-contained files.